lib/travis/build/addons/rethinkdb.rb

```diff
@@ -0,0 +1,44 @@
+require 'shellwords'
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class Rethinkdb < Base
+        SUPER_USER_SAFE = true
+
+        RETHINKDB_GPG_KEY = '0x3A8F2399'
+
+        def after_prepare
+          sh.fold 'rethinkdb' do
+            sh.if "ENV['TRAVIS_OS_NAME'] != 'linux'" do
+              sh.echo "The RethinkDB addon only works on Linux. $TRAVIS_OS_NAME is #{ENV['TRAVIS_OS_NAME']}.", ansi: :red
```

Comment:
The `TaintedOutput` exceptions are coming from here: https://travis-ci.org/travis-ci/travis-build/builds/124611026#L4673

`ENV['TRAVIS_OS_NAME']` is to blame.

Furthermore, at the point when this code is executed, `$TRAVIS_OS_NAME` is not yet defined, so you'll get an empty string there.